### DISCUSSION TITLE
Imprimir informação de tabela com JS.

### DISCUSSION THREAD

**Discussion by Sivazz** [frontendbr/forum#1404] [Perguntas e Respostas]
Pessoal, estou precisando de uma ajuda...

Já queria me adiantar e pedir desculpas se a dúvida não estiver muito bem elaborada ou se for algo muito básico para se perguntar.
Acontece que entrei aqui na empresa como FrontEnd Jr, só manjo de HTML e CSS e meu papel era basicamente pegar os layouts .psd e passar para html/css. Também tenho o básico de Javascript e sei mandar um "Olá mundo" em PHP.

Só que minha chefe, que era fullstack e a responsável pelo site, pediu demissão e a empresa não pretende preencher a vaga dela pois temos um CMS para o site e eles acham que tendo alguém que saiba usar o CMS (eu) já é o suficiente.
Tenho me virado sozinho da forma que dá mas estão começando a surgir algumas demandas que eu não sei muito bem como fazer e o CMS não ajuda muito.

Meu problema é o seguinte:

O site da empresa tem as páginas dos produtos e essas páginas contém alguns detalhes e informações técnicas. Agora nós vamos começar a cadastrar o preço dos produtos.
Para fazer isso atualmente eu tenho que ir até o CMS, abrir a página de cada produto individualmente e inserir o preço lá.

O problema é: Temos mais de 500 produtos cadastrados no nosso site e os valores deles mudam quase que mensalmente, e em algumas ocasiões mudam mais de uma vez ao longo do mês.

Eu queria saber como automatizar isso pois só esse mês os preços alteraram 3 vezes e nas 3 eu precisei fazer esse trampo de ir página a página atualizando tudo.

Eu recebo uma planilha com o modelo do produto, o ID e o preço. Assim:

`	<table border="1px">
  <tr>
    <th>Model</th>
    <th>ID</th> 
    <th>Price</th>
  </tr>
  <tr>
    <td>MBCL</td>
    <td>XQYWER</td>
    <td>R$3.500,00</td>
  </tr>
  <tr>
    <td>MMP1</td>
    <td>WASUPKQ</td>
    <td>R$20.000,00</td>
  </tr>
  <tr>
    <td>MY8SD1</td>
    <td>AFREQWT</td>
    <td>R$6.800,00</td>
  </tr>
</table>`

Minha ideia era desenvolver um código que chamasse as informações dessa tabela, assim só seria necessário atualizar a tabela, e não a página de cada produto uma por uma.

Como eu expliquei no início, sou FrontEnd Jr. no segundo semestre da facul, mas tudo que sei, aprendi por conta própria (a facul não me acrescentou muito até agora).

Queria um norte de vocês, indicações de artigos e oq pesquisar para conseguir fazer o que estou precisando. E não tem problema se não for uma solução simples e imediata, inclusive podem sugerir cursos pois estou vendo que esses problemas vão ficar por minha conta mesmo então vou ter bastante tempo pra quebrar a cabeça estudando formas de resolvê-los. 

Não acho que a faculdade será tão útil nessa parte.

**Comment by felipefabreu** [comment]
Fala Sivaldo, tudo bem?
não tenho muita experiência mas talvez consiga te ajudar numa direção.
Pelo que você falou, parece que as informações dos produtos são estáticas nas páginas, assim, você precisa alterar manualmente página por página quando alguma coisa é modificada. Correto?

Você precisa de um banco de dados para armazenar as informações dos produtos, depois fazer com que cada página puxe as informações desse banco.

Você deve encontrar algumas versões grátis e de fácil aplicação para isso. 

Depois do banco criado, você pode fazer um CRUD(create, read, update e delete) para os produtos, assim permitindo que qualquer um possa atualizar essas informações.

Caso possa ajudar em mais alguma coisa entre em contato.

Boa sorte.

**Comment by kivervinicius** [comment]
O CMS utilizado é proprietário? ou é um Wordpress da vida? caso for e já tiver persistência, poderá atualizar direto no banco (com muito cuidado), caso não for é possível automatizar as tasks(como um macro) passando por todas as telas necessárias por produto.

**Comment by leandroaps** [comment]
Exporta a tabela de produtos como CSV, edita no Excel, e re-importa na base!
Funciona bem com mySQL e SQL, não sei a base que você tá usando ai.

No Excel você pode usar uma PROC pra fazer os updades!
:D

-- 

Leandro A. Siqueira
leandroaps@gmail.com
+55 19 991112239
